core/src/com/unciv/models/ruleset/tile/TileResource.kt

```diff
@@ -24,13 +24,29 @@ class TileResource : RulesetStatsObject() {
     var majorDepositAmount: DepositAmount = DepositAmount()
     var minorDepositAmount: DepositAmount = DepositAmount()
 
-    private val _allImprovements by lazy {
-        if (improvement == null) improvedBy
-        else improvedBy + improvement!!
+    private var improvementsInitialized = false
+    private val allImprovements = mutableSetOf<String>()
+    private var ruleset: Ruleset? = null
+
+    fun getImprovements(): Set<String> {
+        if (improvementsInitialized) return allImprovements
+        val ruleset = this.ruleset
+            ?: throw IllegalStateException("TileResource initialization incomplete")
+        improvement?.also { allImprovements += it }
+        allImprovements.addAll(improvedBy)
+        for (improvement in ruleset.tileImprovements.values) {
+            if (improvement.getMatchingUniques(UniqueType.ImprovesResources).none { matchesFilter(it.params[0]) }) continue
```

Comment:
As a unique, I would expect it to accept tile modifiers, but it doesn't
Best solution is to make it do so and return a different set of improvements based on tile
Second best is to write in doc of unique "does not accept conditionals"